test/test_warnings.py

```diff
@@ -0,0 +1,118 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=unused-argument, unused-variable
+"""Test warning helper."""
+
+from test.base import QiskitExperimentsTestCase
+from qiskit_experiments.warnings import deprecated_init_args, deprecated_class
+from qiskit_experiments.framework import BaseExperiment, ExperimentConfig
+
+
+class TestWarningsHelper(QiskitExperimentsTestCase):
+    """Test case for warnings decorator with tricky behavior."""
+
+    def test_creating_experiment_with_old_args(self):
+        """Test raise warning and update configuration when create instance with old kwargs."""
+
+        @deprecated_init_args(
+            arguments_map={"some_old_arg1": "some_new_arg1", "some_old_arg2": None},
```

Comment:
Fake experiment doesn't have some_old_arg2, so what if you still need to use it? Usually deprecation involves backwards compatibility so we probably need access to these arguments if they are not a simple renaming

Reply:
Sometime this happens since we are still lazy about the deprecation policy.